app/src/main/java/com/ddoczi/tasky/authentication/presentation/login/LoginScreen.kt

```diff
@@ -0,0 +1,90 @@
+package com.ddoczi.tasky.authentication.presentation.login
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.ddoczi.tasky.R
+import com.ddoczi.tasky.authentication.presentation.composables.BaseBackground
+import com.ddoczi.tasky.authentication.presentation.composables.BaseButton
+import com.ddoczi.tasky.authentication.presentation.composables.EmailField
+import com.ddoczi.tasky.authentication.presentation.composables.PasswordField
+import com.ddoczi.tasky.ui.theme.Gray
+import com.ddoczi.tasky.ui.theme.TaskyTheme
+
+@Composable
+fun LoginScreen(
+    viewModel: LoginViewModel = LoginViewModel()
+) {
```

Comment:
I'd avoid passing the ViewModel directly to screen composables. This has 2 major reasons:
1. You're making it harder to have a preview, since you'll need to provide a fake ViewModel instance and fakes for everything the ViewModel needs in its constructor.
2. UI tests are also a bit harder since you'll need to provide these fakes.

What I suggest instead:
```
// In your ViewModel
@HiltViewModel
class SampleViewModel @Inject constructor(): ViewModel() {
    private val _state = MutableStateFlow(SampleState()) // Or Compose state alternatively
    val state = _state.asStateFlow()

    fun onEvent(event: SampleEvent) {
        when(event) {
            is SampleEvent.Event1 -> // Handle event1
            is SampleEvent.Event2 -> // Handle event2
            // ...
        }
    }

    // ...
}

// In your NavHost
composable(Routes.sample) {
    val viewModel = hiltViewModel<SampleViewModel>()
    val state by viewModel.state.collectAsStateWithLifecycle()
    SampleScreen(
        state = state,
        onEvent = viewModel::onEvent
    )
}

// Your screen composable
@Composable
fun SampleScreen(
    state: SampleState,
    onEvent: (SampleEvent) -> Unit
) {
    // ...
    Button(onClick = { onEvent(SampleEvent.ClickedButtonX) } {
        Text("Click me")
    }
}
```